drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1612,26 +1602,52 @@ static int eth_sam_gmac_set_config(struct device *dev,
 	}
 }
 
-static int eth_sam_gmac_get_config(struct device *dev,
+static int eth_sam_gmac_set_config(struct device *dev,
 				   enum ethernet_config_type type,
-				   struct ethernet_config *config)
+				   const struct ethernet_config *config)
+{
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_QAV_PARAM:
+		return eth_sam_gmac_set_qav_param(dev, type, config);
+	default:
```

Comment:
Would look better here if the return statement is outside of switch, then we can also remove the default case.

Reply:
Hmm, we could add `int ret` at the beginning and set it to `-ENOTSUP` after `default`.

Removing the default case is not really an option unless we change the `switch` to a series of `if`s (if that is preferred) - otherwise it would generate warnings for not having cases for all enum values.

Are there some general guidelines on where to use series of ifs vs a switch?

Reply:
Ok, lets add the ret variable then, also default: could just break.
Not sure if there are any guidelines for this. While the code works ok, it just looks a bit weird as if the return statement is missing.

Reply:
Ok sure, will do!